flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -214,7 +219,7 @@ protected void onBarrier(int channelIndex) throws IOException {
 	}
 
 	@Override
-	public boolean processCancellationBarrier(CancelCheckpointMarker cancelBarrier) throws Exception {
+	public void processCancellationBarrier(CancelCheckpointMarker cancelBarrier) throws Exception {
```

Comment:
Refer to this idea https://github.com/apache/flink/pull/11351/files#r397598435, we can also avoid calling `notifyCheckpointCompletedOrCanceled` in five paths.